### PR TITLE
Added extensive new ALS driver

### PR DESCRIPTION
**Pull request by scapri**
Testing inconclusive I don't understand why the ALS driver didn't run

### CODE REVIEW

File: als_driver.c [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+
+#include <stdio.h>
+#include <stdint.h>
+
+int als_read_reg(uint32_t als_reg_addr)
+{
+    printf("read value of 0x%X from als at addr 0x%X\n",
+           200, ALS_READ_ADDR);
+
```
Comment (ciaodog) [inline als_driver.c:12]: all you are doing is printing - what's the point?  You aren't really ready for a review, are you?

**Review by ciaodog — CHANGES_REQUESTED** [review]

File: als_driver.c (hunk 1 above)
Comment (ciaodog) [reply als_driver.c:12]: This still isn't anything but a print - can you give me more information about the purpose of this pull request?

Comment (scapri) [reply als_driver.c:12]: I put those there as a placeholder so the rest of the team can fill in the rest.

File: als_driver.c [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#define ALS_EN_GPIO 11
+#define ALS_READ_ADDR 0xA110FF30
+
+
+#include <stdio.h>
```
Comment (ciaodog) [inline als_driver.c:5]: and what is stdio doing here - our embedded system doesn't support stdio.h

**Review by ciaodog — CHANGES_REQUESTED** [review]

File: als_driver.c (hunk 2 above)
Comment (scapri) [reply als_driver.c:5]: this is for emulation, not for the embedded system.  Once the rest of the team adds their code and tests it, then all of the printfs and stdio will be removed, and replace with the embedded code

Comment (ciaodog) [reply als_driver.c:5]: ok, I understand.

File: als_driver.c (hunk 1 above)
Comment (ciaodog) [reply als_driver.c:12]: Ok, that makes sense

File: als_driver.c [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+#include <stdint.h>
+
+int als_read_reg(uint32_t als_reg_addr)
+{
+    printf("read value of 0x%X from als at addr 0x%X\n",
+           200, ALS_READ_ADDR);
```
Comment (ciaodog) [inline als_driver.c:11]: what's with the magic number here?

**Review by ciaodog — CHANGES_REQUESTED** [review]

File: als_driver.c (hunk 3 above)
Comment (scapri) [reply als_driver.c:11]: Oh yeah, I need to fix that...

Comment (ciaodog) [reply als_driver.c]: Ok, this fix looks good.

**Review by ciaodog — APPROVED** [review]

File: als_driver.c (hunk 3 above)
Comment (ciaodog) [reply als_driver.c]: oh wait, shouldn't this actually be reading the h/w, not faking out a constant value?

Comment (scapri) [reply als_driver.c]: Wouldn't it be nice if I could resolve this comment....

**Review by ciaodog — CHANGES_REQUESTED** [review]
more changes needed

**Pull request merged**